### PR TITLE
Lots of fixes to test framework for Firefox.

### PR DESCRIPTION
**Pull request by mithro**

### CODE REVIEW

**Comment by mithro** [comment]
PTAL.

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,4 @@
 *.log*
 tools/*
 .testrepository
+*.png
```
Comment (dstoc) [inline .gitignore:10]: ?

Comment (mithro) [inline .gitignore:10]: The run-tests.py generates png screenshot files. This tells git to ignore them....

File: run-tests.py [hunk 2]
```diff
@@ -494,13 +499,15 @@ def do_POST(self):
 # Start up a virtual display, useful for testing on headless servers.
 # -----------------------------------------------------------------------------
 
+VIRTUAL_SIZE=(1024, 2000)
```
Comment (dstoc) [inline run-tests.py:502]: 2000 seems quite large?

File: run-tests.py [hunk 3]
```diff
@@ -554,6 +565,10 @@ def directory_cleanup(directory, shutil=shutil):
 
 elif args.browser == "Firefox":
     driver_arguments['firefox_profile'] = webdriver.FirefoxProfile()
+    # Firefox will often pop-up a dialog saying "script is taking too long" or
+    # similar. So we can notice this problem we use "accept" rather then the
+    # default "dismiss".
+    webdriver.DesiredCapabilities.FIREFOX["unexpectedAlertBehaviour"] = "accept"
```
Comment (dstoc) [inline run-tests.py:571]: Neat.

File: .gitignore (hunk 1 above)
Comment (dstoc) [inline .gitignore:10]: Maybe it could put them somewhere else. It would be nice not to ignore globally.

But doesn't really matter.

**Comment by dstoc** [comment]
lgtm

File: run-tests.py (hunk 2 above)
Comment (mithro) [inline run-tests.py:502]: I started with 768 and increased it until I got most of the error output in the screenshot.

**Pull request merged**